pandas/io/pytables.py

```diff
@@ -2771,7 +2765,11 @@ def read(self, **kwargs):
     def write(self, obj, **kwargs):
         super(SeriesFixed, self).write(obj, **kwargs)
         self.write_index('index', obj.index)
-        self.write_array('values', obj.values)
+        if is_datetime64tz_dtype(obj.dtype):
```

Comment:
this is reaching into the internal implementation way too much.

you can use ``obj.get_values()`` here

Reply:
The problem with `obj.get_values()` is that it returns an `array` with dtype `datetime64[ns]`, but we need a `DatetimeIndex` with dtype `datetime64[ns, tz]` to preserve the timezone information. I tried using `obj.dt.values`, which worked on `0.23.0`, but not on `master`.

Reply:
so ``write_array`` already handles the objects corrctly, try just passing ``obj`` into that (you *may* need to add a case in write_array, but see where this gets you